pkg/asset/cluster/tfvars.go

```diff
@@ -385,9 +385,13 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 		if err != nil {
 			return err
 		}
+
+		var masterSpecs []*openstackprovider.OpenstackProviderSpec
+		for _, master := range masters {
+			masterSpecs = append(masterSpecs, master.Spec.ProviderSpec.Value.Object.(*openstackprovider.OpenstackProviderSpec))
+		}
 		data, err = openstacktfvars.TFVars(
-			masters[0].Spec.ProviderSpec.Value.Object.(*openstackprovider.OpenstackProviderSpec),
-			installConfig.Config.Platform.OpenStack.Cloud,
+			masterSpecs,
```

Comment:
Probably this is a stupid question, but is it possible to have more than one master spec in our case?

Reply:
no, there can only be one control plane machinepool

Reply:
in theory, we should be creating a spec per AZ

Reply:
Yep, but it always creates only one spec: https://github.com/openshift/installer/pull/3918/files#diff-a2bd450b1cd3199485e614928c597a91R48

Reply:
looks like a bug

Reply:
I believe I fixed this now. Do double check me though